plugins/kitpvp-paper/src/main/java/dev/drekamor/warp/database/DatabaseManager.java

```diff
@@ -0,0 +1,125 @@
+package dev.drekamor.warp.database;
+
+import dev.drekamor.warp.util.Warp;
+import net.civmc.kitpvp.KitPvpPlugin;
+import vg.civcraft.mc.civmodcore.dao.ManagedDatasource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DatabaseManager {
+    private final KitPvpPlugin plugin;
+    private final ManagedDatasource dataSource;
+
+    public DatabaseManager (KitPvpPlugin plugin, ManagedDatasource dataSource) {
+        this.plugin = plugin;
+        this.dataSource = dataSource;
+        this.plugin.info("Initialised a database connection");
+
+
+        initialiseTables();
+        dataSource.updateDatabase();
+    }
+
+    private void initialiseTables() {
+        dataSource.registerMigration(2, false, """
+            CREATE TABLE IF NOT EXISTS warps (
+                name VARCHAR(64) PRIMARY KEY,
+                world VARCHAR(64) NOT NULL,
+                x REAL NOT NULL,
+                y REAL NOT NULL,
+                z REAL NOT NULL,
+                pitch REAL NOT NULL,
+                yaw REAL NOT NULL,
+                gamemode VARCHAR(16) NOT NULL
+            )
+            """);
+
+    }
+
+    public Map<String, Warp> getWarps() {
+
+        Map<String, Warp> warps = new HashMap<>();
+        try (Connection connection = this.dataSource.getConnection()){
+            PreparedStatement statement = connection.prepareStatement("SELECT * FROM warps");
+            ResultSet results = statement.executeQuery();
+            while(results.next()) {
+                warps.put(
+                    results.getString("name"),
+                    new Warp(
+                        results.getString("name"),
+                        results.getString("world"),
+                        results.getDouble("x"),
+                        results.getDouble("y"),
+                        results.getDouble("z"),
+                        results.getFloat("pitch"),
+                        results.getFloat("yaw"),
+                        results.getString("gamemode")
+                    )
+                );
+            }
+            connection.close();
```

Comment:
You don't need to call close here. The connection is automatically closed with try-with-resources